infrastructure-scripts/Get-SqlDbServiceAccount.ps1

```diff
@@ -0,0 +1,144 @@
+    [Parameter(Mandatory = $true)]
+    [String]  $DataBaseName,
+    [Parameter(Mandatory = $true)]
+    [String]$SqlUserName,
+    [Parameter(Mandatory = $true)]
+    [SecureString]  $SqlPassword,
+    [Parameter(Mandatory = $true)]
+    [String]  $SqlServiceAccountName,
+    [Parameter(Mandatory = $true)]
+    [String]  $SQLResourceGroup,
+    [Parameter(Mandatory = $true)]
+    [String]  $Enviroment,
+    [Parameter(Mandatory = $true)]
+    [String] $KeyVaultName
+)
+
+$ErrorActionPreference = 'Stop'
+
+
+function New-AzureSQLServerFirewallRule {
+    $agentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    New-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+function Update-AzureSQLServerFirewallRule {
+    $agentIP = (New-Object net.webclient).downloadstring("http://checkip.dyndns.com") -replace "[^\d\.]"
+    Set-AzureRmSqlServerFirewallRule -StartIPAddress $agentIp -EndIPAddress $agentIp -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+}
+
+function Get-RandomPassword {
+    $Password = ([char[]]([char]33..[char]95) + ([char[]]([char]97..[char]126)) + 0..9 | Sort-Object { Get-Random })[0..33] -join ''
+    return $Password
+}
+function Invoke-SqlQuery([String] $query) {
+    $output = Invoke-Sqlcmd -ServerInstance $ServerFQDN -Database $DataBaseName -Username $SqlUserName -Password $SqlPassword -OutputSqlErrors $true -Query $query
+    return $output
+}
+
+function Remove-AzureSQLServerFirewallRule {
+    If ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue)) {
+        Remove-AzureRmSqlServerFirewallRule -FirewallRuleName $AzureFirewallName -ServerName $ServerName -ResourceGroupName $SQLResourceGroup
+    }
+}
+
+try {
+    $AccountExistQuery = @"
+                        SELECT * FROM sys.database_principals WHERE name = '$SqlServiceAccountName'
+"@
+
+    $secretName = "${Enviroment}-${SqlServiceAccountName}"
+
+    Write-Host "$secretName"
+
+    If ((Get-AzureRmSqlServerFirewallRule -ServerName $ServerName -ResourceGroupName $SQLResourceGroup -FirewallRuleName $AzureFirewallName -ErrorAction SilentlyContinue) -eq $null) {
+        New-AzureSQLServerFirewallRule
+    }
+    else {
+        Update-AzureSQLServerFirewallRule
+    }
+
+    Write-Host "$ServerName.database.windows.net"
+    $ServerFQDN = "$ServerName.database.windows.net"
+    $accountExist = Invoke-SqlQuery $AccountExistQuery
+    Write-Output $accountExist
+
+
+    if ($accountExist) {
+        $AccountPassword = Get-AzureKeyVaultSecret -VaultName $KeyVaultName -name $secretName
+        $query = @"
+        ALTER USER "$SqlServiceAccountName" WITH PASSWORD = '$AccountPassword'
```

Comment:
I'm unsure if we should be resetting the password every deploy here. We don't know how some apps will behave when the connection breaks for a few seconds.

Reply:
Ok Removed Section.